iodata/formats/wfx.py

```diff
@@ -18,224 +18,156 @@
-                       '</Number of Occupied Molecular Orbitals>'],
-        'num_perturbations': ['<Number of Perturbations>',
-                              '</Number of Perturbations>'],
-        'num_electrons': ['<Number of Electrons>',
-                          '</Number of Electrons>'],
-        'num_alpha_electron': ['<Number of Alpha Electrons>',
-                               '</Number of Alpha Electrons>'],
-        'num_beta_electron': ['<Number of Beta Electrons>',
-                              '</Number of Beta Electrons>'],
-        'spin_multi': ['<Electronic Spin Multiplicity>',
-                       '</Electronic Spin Multiplicity>']
+    labels_number = {
+        # integer numbers
+        '<Number of Nuclei>': 'num_atoms',
+        '<Number of Occupied Molecular Orbitals>': 'num_occ_mo',
+        '<Number of Perturbations>': 'num_perturbations',
+        '<Number of Electrons>': 'num_electrons',
+        '<Number of Core Electrons>': 'num_core_electrons',
+        '<Number of Alpha Electrons>': 'num_alpha_electron',
+        '<Number of Beta Electrons>': 'num_beta_electron',
+        '<Number of Primitives>': 'num_primitives',
+        '<Electronic Spin Multiplicity>': 'spin_multi',
+        # float numbers
+        '<Net Charge>': 'charge',
+        '<Energy = T + Vne + Vee + Vnn>': 'energy',
+        '<Virial Ratio (-V/T)>': 'virial_ratio',
+        '<Nuclear Virial of Energy-Gradient-Based Forces on Nuclei, W>': 'nuc_viral',
+        '<Full Virial Ratio, -(V - W)/T>': 'full_virial_ratio',
+    }
+    labels_array = {
+        '<Nuclear Cartesian Coordinates>': 'atcoords',
+        '<Nuclear Charges>': 'nuclear_charge',
+        '<Atomic Numbers>': 'atnums',
+        '<Primitive Exponents>': 'exponents',
+        '<Primitive Centers>': 'centers',
+        '<Primitive Types>': 'types',
+        '<Molecular Orbital Energies>': 'mo_energy',
+        '<Molecular Orbital Occupation Numbers>': 'mo_occ',
+        '<Molecular Orbital Primitive Coefficients>': 'mo_coeff',
+    }
+    labels_other = {
+        '<Nuclear Names>': 'nuclear_names',
+        '<Molecular Orbital Spin Types>': 'mo_spin',
+        '<Nuclear Cartesian Energy Gradients>': 'nuclear_gradient',
     }
 
-    dict_int = {}
-    for key, val in int_label.items():
-        int_info = _helper_section(f_content=f_content,
-                                   start=val[0],
-                                   end=val[1])
-        if int_info:
-            dict_int[key] = np.array(int_info[0], dtype=np.int)
+    # load raw data
+    data = parse_wfx(lit)
+    # process raw data
+    result = {}
+    for key, value in data.items():
+        if key in labels_str:
+            result[labels_str[key]] = value[0]
+        elif key in labels_number:
+            result[labels_number[key]] = np.array(eval(value[0]))
+        elif key in labels_array:
+            breakdown = [eval(item) for line in value for item in line.split()]
```

Comment:
Idem: better to avoid calling eval on data from external sources. The solution here is similar, i.e. splitting `labels_array` into `labels_array_int` and `labels_array_float`.

Instead of the list comprehension, also `np.fromstring(line, dtype=..., sep=" ")` can be used. See https://mail.google.com/mail/u/0/#inbox That would be a lot faster for large arrays.

Reply:
Fixed.